server/utils/cloud/infrastructure_confirmation.py

```
@@ -29,56 +43,110 @@ def _send_ws(payload: dict, tool_name: str):
-        
-        # Find the pending confirmation
+            update_workflow_websocket_context(user_id, session_id)
+
+        # Resolve the waiter.
         if confirmation_id in _pending_confirmations:
             confirmation_data = _pending_confirmations[confirmation_id]
             confirmation_data['result'] = decision
+            edited = data.get('edited_patterns')
+            if isinstance(edited, dict):
+                confirmation_data['edited_patterns'] = edited
             logger.debug(f"WEBSOCKET: Confirmation {confirmation_id} resolved with decision: {decision}")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)